manifests/init.pp

```diff
@@ -666,6 +664,10 @@
     $_purge_mutators   = $full_purge_hash['mutators']
   }
 
+  # (#463) This well-known anchor serves as a reference point so all checks are
+  # managed after all plugins.  It must always exist in the catalog.
+  anchor { 'plugins_before_checks': }
```

Comment:
I thought about using the spaceship operator but didn't because it's very likely some end users will have virtual or exported check resources.  Collections realize resources, which would be a PITA for the end users.

This isn't the full blown anchor pattern, which is more about class containment.  This is more of a lightweight, "Here's a single, well-known reference point to establish before and after."

I chatted with some people in the Puppet slack, and the best alternative to the anchor idea was https://github.com/binford2k/binford2k-manifold, but I did get some validating feedback a single anchor is reasonable in this situation and I like how simple it is to explain.  All plugins are managed before the anchor, all checks are managed after the anchor, the anchor always exists in the catalog.